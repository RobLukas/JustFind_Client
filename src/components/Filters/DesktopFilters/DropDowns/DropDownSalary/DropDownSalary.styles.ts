import styled from 'styled-components';
import Slider from '@material-ui/core/Slider';

const DropDownSalary = styled.div`
  position: relative;
`;

const Wrapper = styled.div`
  position: absolute;
  background-color: ${({ theme }) => theme.styles.secondary.background};
  width: 200px;
  box-shadow: 0px 2px 24px 0px rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  margin-top: 5px;
  padding: 12px 16px;
  z-index: 1;
  left: 50%;
  right: 0;
  transform: translate(-50%, 0);
`;

const Title = styled.div`
  color: ${({ theme }) => theme.styles.filter.button.color};
  padding-bottom: 20px;
  font-size: 14px;
`;

const SliderMui = styled(Slider)`
  &.MuiSlider-root {
    color: #ff4081;
  }
`;

export default {
  DropDownSalary,
  Wrapper,
  Title,
  SliderMui,
};
