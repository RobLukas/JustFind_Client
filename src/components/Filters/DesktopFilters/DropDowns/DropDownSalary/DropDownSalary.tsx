import React, { FC, useCallback, useState } from 'react';

import S from './DropDownSalary.styles';
import { SalaryButton } from 'components/common';
import { ClickAwayListener } from '@material-ui/core';

const DropDownSalary: FC = () => {
  const [open, setOpen] = useState(false);

  const handleClick = useCallback(() => {
    setOpen(!open);
  }, [open]);

  const handleClickAway = useCallback(() => {
    setOpen(false);
  }, []);

  return (
    <S.DropDownSalary>
      <SalaryButton handleClick={handleClick} />
      {open ? (
        <ClickAwayListener onClickAway={handleClickAway}>
          <S.Wrapper>
            <S.Title>{'Choose salary range'}</S.Title>
            <S.SliderMui />
          </S.Wrapper>
        </ClickAwayListener>
      ) : null}
    </S.DropDownSalary>
  );
};

export default DropDownSalary;
