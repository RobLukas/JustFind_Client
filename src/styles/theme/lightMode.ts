export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
