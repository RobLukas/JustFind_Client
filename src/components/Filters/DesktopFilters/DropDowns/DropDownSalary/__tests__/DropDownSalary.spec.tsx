import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';

import lightMode from 'styles/theme/lightMode';
import { ThemeProvider } from 'styled-components';
import DropDownSalary from '../DropDownSalary';

describe('<DropDownSalary />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <DropDownSalary />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
