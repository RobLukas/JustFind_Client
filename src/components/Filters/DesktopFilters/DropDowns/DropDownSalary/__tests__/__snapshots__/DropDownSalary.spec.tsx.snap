// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownSalary /> render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  margin: 0 10px 0 0;
  display: inline-block;
  position: relative;
}

.c2 {
  color: #777;
  background: #fff;
  margin-left: 0;
  cursor: pointer;
  padding: 0 7px 0 12px;
  margin: 0;
  box-shadow: none;
  text-transform: none;
  height: 35px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.8rem;
  border-radius: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  outline: 0;
  border: 0;
  -webkit-tap-highlight-color: transparent;
}

.c2.MuiSvgIcon-root {
  fill: #777;
}

.c2:hover {
  background-color: rgba(255,255,255,0.08);
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
}

.c4.MuiSvgIcon-root {
  font-size: 1.2rem;
  line-height: inherit;
}

.c3.MuiSvgIcon-root {
  font-size: 1rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root c3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
        />
      </svg>
      Salary
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root c4"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
